roxctl/summaries/policy/policy.go

```diff
@@ -45,16 +51,29 @@ func policySeverityFromString(s string) Severity {
 	}
 }
 
+func checkNetworkPolicyField(p *storage.Policy) bool {
+	for _, section := range p.GetPolicySections() {
+		for _, group := range section.GetPolicyGroups() {
+			if _, ok := networkPolicyFields[group.GetFieldName()]; ok {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 // NewPolicySummaryForPrinting creates a Result that shall be used for printing and holds
 // all relevant information regarding violated policies, failing policies and a summary of all violated policies
 // by severity
 // NOTE: The returned *Result CAN be passed to json.Marshal
-func NewPolicySummaryForPrinting(alerts []*storage.Alert, forbiddenEnforcementAction storage.EnforcementAction) *Result {
+func NewPolicySummaryForPrinting(alerts []*storage.Alert, forbiddenEnforcementAction storage.EnforcementAction) (*Result, bool) {
 	entityMetadataMap := createEntityMetadataMap(alerts)
 	numOfSeveritiesByEntities := createNumOfSeverityByEntity(entityMetadataMap)
 	numOfSeveritiesAcrossEntities := createNumOfSeverityMap()
 	policiesByEntity := make(map[string]map[string]*Policy, len(entityMetadataMap))
 
+	foundNetworkPolicyFields := false
```

Comment:
I'm not sure this should be done at this specific place, it doesn't match at all what `NewPolicySummaryForPrinting` is supposed to do.
Would it be possible to factor this out either:
- into a separate utils function, which would mean we would iterate through `alerts` twice
- provide a generic way to provide i.e. options to include additional information based on the policy, something like:
```go
type newResult struct {
  res Result
  hasNetworkPolicyField bool
}
NewPolicySummaryForPrinting(alerts []*storage.Alert, forbiddenEnforcementAction storage.EnforcementAction, WithCheckForNetworkPolicyField()) *newResult
```
I fear if we leave it like this now and the next requirement for something like this comes up we will add flag over flag to the function as return value.

Reply:
You are right. I think it would be cleaner to separate this into a different function to remove alerts with network policy fields.